quartz/plugins/emitters/componentResources.ts

```diff
@@ -172,59 +179,114 @@ export const ComponentResources: QuartzEmitterPlugin<Options> = (opts?: Partial<
       return []
     },
     async emit(ctx, _content, resources): Promise<FilePath[]> {
+      const promises: Promise<FilePath>[] = []
+      const cfg = ctx.cfg.configuration
       // component specific scripts and styles
       const componentResources = getComponentResources(ctx)
       // important that this goes *after* component scripts
       // as the "nav" event gets triggered here and we should make sure
       // that everyone else had the chance to register a listener for it
 
-      if (fontOrigin === "googleFonts") {
-        resources.css.push(googleFontHref(ctx.cfg.configuration.theme))
-      } else if (fontOrigin === "local") {
+      let googleFonts = null
+      if (fontOrigin === "local") {
         // let the user do it themselves in css
+      } else if (fontOrigin === "googleFonts") {
+        if (cfg.theme.cdnCaching) {
+          resources.css.push(googleFontHref(cfg.theme))
+        } else {
+          let match
+          const bufferPromise: Promise<FontContent>[] = []
+
+          const regex = /url\((https:\/\/fonts.gstatic.com\/s\/[^)]+\.(woff2|ttf))\)/g
+
+          googleFonts = await fetch(googleFontHref(ctx.cfg.configuration.theme)).then((res) =>
+            res.text(),
+          )
```

Comment:
```suggestion
          googleFonts = (await fetch(googleFontHref(ctx.cfg.configuration.theme))).text()
```

generally avoid mixing `await` and `.then()` / `.catch()`

Reply:
is there an article or certain practice within ts about this?

I assume this has to do with how js manage its event loop?